addons/spreadsheet/static/src/hooks.js

```diff
@@ -0,0 +1,86 @@
+/** @odoo-module */
+
+import { useEffect, useExternalListener, useState } from "@odoo/owl";
+import { loadBundle } from "@web/core/assets";
+
+/**
+ *
+ * @param {() => Model | undefined} model
+ * @returns
+ */
+export function useSpreadsheetPrint(model) {
+    let frozenPrintState = undefined;
+    const printState = useState({ active: false });
+
+    useExternalListener(
+        window,
+        "keydown",
+        async (ev) => {
+            const isMeta = ev.metaKey || ev.ctrlKey;
+            if (ev.key === "p" && isMeta) {
+                if (!model()) {
+                    return;
+                }
+                ev.preventDefault();
+                ev.stopImmediatePropagation();
+                await beforePrint();
+            }
+        },
+        { capture: true }
+    );
```

Comment:
this is not enough: using the print from the menu will not set the print zone correctly. I think you can use the event "beforeprint" https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeprint_event
However I don't know how it will handle the async work...

Reply:
Cannot do that unfortunately, `beforeprint` does not support async callbacks. We cannot await a state change & rerendering of components (at least ATM).